### PR TITLE
upload when filesize is not 0

### PR DESCRIPTION
**Pull request by jasonwee**
This is for issue #59 , before actual upload, we check if the file is empty. if it is, don't have to upload

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
The proposed change would work, but I would suggest a slightly different approach.

Note that the `CheckStateTask` calls `Utils.isExcluded()` to avoid uploading files and directories with specific names. How about extending this method in `StorjUtil`, i.e. create `StorjUtil.isExcluded()` that calls `Utils.isExcluded()` and check in addition if the file size is 0? Then you just need to call the new `StorjUtil.isExcluded()` instead of `Utils.isExcluded()` in `CheckStateTask`.

You will also have to replace this call in `StorjOverlayIconProvider` to show proper icon (i.e. no icon) for empty files, which is not currently covered in this PR.

Please, also add a few test cases that cover the empty file use case:
- in `OverlayHelperTest` check that empty file has the `NONE` overlay icon - similar to the `excludedFile()` test case
- in `CheckStateTaskTest` check that empty file is not uploaded - similar to the `excludedFileNoCloud()` test case.

**Comment by jasonwee** [comment]
> Note that the CheckStateTask calls Utils.isExcluded() to avoid uploading files and directories with specific names. How about extending this method in StorjUtil, i.e. create StorjUtil.isExcluded() that calls Utils.isExcluded() and check in addition if the file size is 0? Then you just need to call the new StorjUtil.isExcluded() instead of Utils.isExcluded() in CheckStateTask.

Yes, I have made changes to that. But on line, https://github.com/GooBox/goobox-sync-storj/blob/dbba432a81b98db66520df902a1880e5cef57064/src/main/java/io/goobox/sync/storj/CheckStateTask.java#L143, https://github.com/GooBox/goobox-sync-storj/blob/dbba432a81b98db66520df902a1880e5cef57064/src/main/java/io/goobox/sync/storj/CheckStateTask.java#L180 and https://github.com/GooBox/goobox-sync-storj/blob/dbba432a81b98db66520df902a1880e5cef57064/src/main/java/io/goobox/sync/storj/CheckStateTask.java#L263 of the class `CheckStateTask`, does your suggestion covered that cases?

**Comment by kaloyan-raev** [comment]
You are right, those are not covered. I was thinking only about the case where you create a new local empty file.

Those other cases are for make an existing file empty. But what would you do in those cases? Simply avoiding the upload will leave the old version of the file (with some content) on the Storj cloud. Should you delete it from the cloud in this case?

**Comment by jasonwee** [comment]
> Those other cases are for make an existing file empty. But what would you do in those cases? Simply avoiding the upload will leave the old version of the file (with some content) on the Storj cloud. Should you delete it from the cloud in this case?

Good questions.

We will probably get the error in the log but the bridge will stop it from uploading. So this really isn't a big issue. But now I understand why you made this changed only on line 185 as a mean for moving forward.

File: src/main/java/io/goobox/sync/storj/StorjUtil.java [hunk 1]
```diff
@@ -71,4 +73,16 @@ public static boolean isTemporaryError(int code) {
         }
     }
 
+    public static boolean isExcluded(Path path) throws IOException {
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/StorjUtil.java]: It would be better if you catch the IOException here instead of re-throwing it. If you catch IOException, perhaps return true, i.e. to exclude the file from sync. If the file size cannot be determined then most probably the sync will fail too, so better exclude that file.

Comment (jasonwee) [reply src/main/java/io/goobox/sync/storj/StorjUtil.java]: yea, sure
